framework_agreement_sourcing/model/logistic_requisition.py

```diff
@@ -36,18 +32,21 @@ def _generate_default_source(self):
         Note: older versions of the module had logic to automatically choose
         one or more agreements.
         """
+        Agreement = self.env['product.pricelist']
         _super = super(LogisticsRequisitionLine, self)
         new_source = _super._generate_default_source()
 
-        Agreement = self.env['framework.agreement']
-        ag_domain = Agreement.get_agreement_domain(
-            self.product_id.id,
-            self.requested_qty,
-            None,
-            self.requisition_id.date,
-            self.requisition_id.incoterm_id.id,
-            self.requisition_id.incoterm_address,
+        req_date = datetime.strptime(self.requisition_id.date, DATETIME_FORMAT)
```

Comment:
here you should use fields.Datetime.from_string()
https://www.odoo.com/documentation/8.0/reference/orm.html#openerp.fields.Datetime